File: EinkaufszettelApp/app/src/main/java/com/example/einkaufszettelapp/Produkt.java
@Entity(foreignKeys = @ForeignKey(
        entity = Kategorie.class,
        parentColumns = "id",
        childColumns = "kategorieId",
        onDelete = ForeignKey.CASCADE
))
public class Produkt {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @NonNull
    public String name;

    public int kategorieId;
}

File: Einkaufszettel2/app/src/main/java/com/example/einkaufszettel/Produkt.kt
package com.example.einkaufszettel.data

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    foreignKeys = [
        ForeignKey(
            entity = Kategorie::class,
            parentColumns = ["id"],
            childColumns = ["kategorieId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Produkt(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    val name: String,
    val kategorieId: Int
)
